pkg/resources/api_authentication_integration_with_authorization_code_grant.go

```diff
@@ -0,0 +1,523 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/collections"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/logging"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var apiAuthAuthorizationCodeGrantSchema = func() map[string]*schema.Schema {
+	uniq := map[string]*schema.Schema{
+		"oauth_authorization_endpoint": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "unknown",
+			Description: "Specifies the URL for authenticating to the external service.",
+		},
+		"oauth_allowed_scopes": {
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Optional:    true,
+			Description: "Specifies a list of scopes to use when making a request from the OAuth by a role with USAGE on the integration during the OAuth client credentials flow.",
+		},
+		"oauth_grant": {
```

Comment:
I think we had this discussion during SDK implementation and agreed to include this field. Let's keep it for now.

Reply:
Hmm, adding it wouldn't be a breaking change, but removing will, so I'm wondering if that should be here if it only can have one value (It can be multiple values in Snowflake, but here we expect the exact type). In the V1 resource guidelines we wanted to prevent such fields from being added, so maybe it should be removed?
cc: @sfc-gh-asawicki (wdyt?)

Reply:
Done.